File: frontend/src/types/medicine.ts
import type { Maybe } from "yup";

export interface Medicine {
  id: string;
  name: string;
  quantity: number;
  expiryDate: string;
  brand: string;
  price: number;
  notes?: string;
}

export interface CreateMedicineRequest{
  name: string;
  quantity: number;
  expiryDate: string;
  brand: string;
  price: number;
  notes?: string;
}
export interface CreateMedicineResponse{
  id: string;
}
export interface UpdateMedicineRequest{
  name: string;
  quantity: number;
  expiryDate: string;
  brand: string;
  price: number;
  notes?: string;
}

export interface PaginationResponse<T>{
    pageIndex: number;
    pageSize: number;
    totalCount: number; 
    items: T[];
}
export interface GetAllMedicinesResponse{
  data: PaginationResponse<Medicine>;
}

export interface MedicineState {
  loading: boolean;
  error?: string;
  medicine?: Medicine;
}

File: backend/Services/Medicine/Medicine.Domain/Entities/Medicine.cs
using Medicine.Domain.ValueObjects;

namespace Medicine.Domain.Entities;

public class Medicine
{
    public MedicineId Id { get; private set; } 
    public string Name { get; private set; }
    public DateOnly ExpiryDate { get; private set; }
    public int Quantity { get; private set; }
    public decimal Price { get; private set; }
    public string Brand { get; private set; }
    public string? Notes { get; private set; }

    public Medicine()
    {
        //for EF Core
    }

    private Medicine(MedicineId id, string name, DateOnly expiryDate, int quantity, decimal price, string brand, string? notes)
    {
        Id = id;
        Name = name;
        ExpiryDate = expiryDate;
        Quantity = quantity;
        Price = price;
        Brand = brand;
        Notes = notes;
    }
    public static Medicine Create(string name, DateOnly expiryDate, int quantity, decimal price, string brand,string? notes = default)
    {
        return new(MedicineId.Create(), name, expiryDate, quantity, price,brand,notes);
    }
    public static Medicine Update(MedicineId id, string name, DateOnly expiryDate, int quantity, decimal price, string brand, string? notes = default)
    {
        return new(id, name, expiryDate, quantity, price, brand, notes);
    }
}
